packages/ketcher-react/src/script/editor/operations/simpleObject.ts

```diff
@@ -0,0 +1,337 @@
+    bottomRightX.x = current.x
+    console.log(1)
+  }
+
+  if (current.y < bottomRightY.y && (firstQuarter || secondQuarter)){
+    topLeftY.y = current.y
+    console.log(2)
+  }
+  if (current.x < bottomRightX.x && (secondQuarter || thirdQuarter)){
+    topLeftX.x = current.x
+    console.log(3)
+  }
+  if (current.y > topLeftY.y && (thirdQuarter || forthQuarter)){
+    bottomRightY.y = current.y
+    console.log(4)
+  }
+}
+
+function handleRectangleChangeWithAnchor(item, anchor, current) {
+  const previousPos0 = item.pos[0].get_xy0()
+  const previousPos1 = item.pos[1].get_xy0()
+
+  if (tfx(anchor.x) === tfx(item.pos[1].x)) {
+    item.pos[1].x = anchor.x = current.x
+    current.x = previousPos1.x
+  }
+  if (tfx(anchor.y) === tfx(item.pos[1].y)) {
+    item.pos[1].y = anchor.y = current.y
+    current.y = previousPos1.y
+  }
+  if (tfx(anchor.x) === tfx(item.pos[0].x)) {
+    item.pos[0].x = anchor.x = current.x
+    current.x = previousPos0.x
+  }
+  if (tfx(anchor.y) === tfx(item.pos[0].y)) {
+    item.pos[0].y = anchor.y = current.y
+    current.y = previousPos0.y
+  }
+}
+
+export class SimpleObjectResize extends Base {
+  data: SimpleObjectResizeData
+
+  constructor(
+    id: string,
+    d: any,
+    current: Vec2,
+    anchor: any,
+    noinvalidate: boolean,
+    toCircle: boolean
+  ) {
+    super(OperationType.SIMPLE_OBJECT_RESIZE)
+    this.data = { id, d, current, anchor, noinvalidate, toCircle }
+  }
+
+  execute(restruct: any): void {
+    const struct = restruct.molecule
+    const id = this.data.id
+    const d = this.data.d
+    const current = this.data.current
+    const item = struct.simpleObjects.get(id)
+    const anchor = this.data.anchor
+    if (item.mode === SimpleObjectMode.ellipse) {
+      if (anchor) {
+        if (
+          tfx(anchor.y) !== tfx(item.pos[0].y) &&
+          tfx(anchor.x) !== tfx(item.pos[0].x) &&
+          tfx(anchor.y) !== tfx(item.pos[1].y) &&
+          tfx(anchor.x) !== tfx(item.pos[1].x)
+        ) {
+          handleEllipseChangeIfAnchorIsOnDiagonal(item, current)
+        } else {
+          handleEllipseChangeIfAnchorIsOnAxis(anchor, item, current)
+        }
+      } else if (this.data.toCircle) {
+        const previousPos1 = item.pos[1].get_xy0()
+        const circlePoint = makeCircleFromEllipse(item.pos[0], current)
+        item.pos[1].x = circlePoint.x
+        item.pos[1].y = circlePoint.y
+        this.data.current = previousPos1
+      } else {
+        const previousPos1 = item.pos[1].get_xy0()
+        item.pos[1].x = current.x
+        item.pos[1].y = current.y
+        this.data.current = previousPos1
+      }
+    } else if (item.mode === SimpleObjectMode.line && anchor) {
+      const previousPos1 = anchor.get_xy0()
+      anchor.x = current.x
+      anchor.y = current.y
+      this.data.current = previousPos1
+    } else if (item.mode === SimpleObjectMode.rectangle && anchor) {
+      handleRectangleChangeWithAnchor.call(this, item, anchor, current)
```

Comment:
Why  do you use .call() ? I see handleRectangleChangeWithAnchor does not require any information  from SimpleObject instance, it works with input arguments only.

Reply:
fixed